src/shared/vscode/window.ts

```diff
@@ -0,0 +1,80 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as vscode from 'vscode'
+
+/**
+ * Components associated with {@link vscode.window}.
+ */
+export interface Window {
+    statusBar: StatusBar
+    inputBox: InputBox
+    message: Message
+    progress: Progress
+    dialog: Dialog
+}
+
+/**
+ * Actions associated with status bars in {@link vscode.window}.
+ */
+export interface StatusBar {
```

Comment:
Consider supporting the equivalent of `createStatusBarItem`, since we aren't bound to just one status bar.